Mathlib/LinearAlgebra/QuadraticForm/Isometry.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2020 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying, Eric Wieser
+
+! This file was ported from Lean 3 source module linear_algebra.quadratic_form.isometry
+! leanprover-community/mathlib commit 14b69e9f3c16630440a2cbd46f1ddad0d561dee7
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.LinearAlgebra.QuadraticForm.Basic
+
+/-!
+# Isometries with respect to quadratic forms
+
+## Main definitions
+
+* `quadratic_form.isometry`: `linear_equiv`s which map between two different quadratic forms
+* `quadratic_form.equvialent`: propositional version of the above
+
+## Main results
+
+* `equivalent_weighted_sum_squares`: in finite dimensions, any quadratic form is equivalent to a
+  parametrization of `quadratic_form.weighted_sum_squares`.
+-/
+
+
+variable {ι R K M M₁ M₂ M₃ V : Type _}
+
+namespace QuadraticForm
+
+variable [Semiring R]
+
+variable [AddCommMonoid M] [AddCommMonoid M₁] [AddCommMonoid M₂] [AddCommMonoid M₃]
+
+variable [Module R M] [Module R M₁] [Module R M₂] [Module R M₃]
+
+/-- An isometry between two quadratic spaces `M₁, Q₁` and `M₂, Q₂` over a ring `R`,
+is a linear equivalence between `M₁` and `M₂` that commutes with the quadratic forms. -/
+-- Porting note: not implemented @[nolint has_nonempty_instance]
+structure Isometry (Q₁ : QuadraticForm R M₁) (Q₂ : QuadraticForm R M₂) extends M₁ ≃ₗ[R] M₂ where
+  map_app' : ∀ m, Q₂ (toFun m) = Q₁ m
+#align quadratic_form.isometry QuadraticForm.Isometry
+
+/-- Two quadratic forms over a ring `R` are equivalent
+if there exists an isometry between them:
+a linear equivalence that transforms one quadratic form into the other. -/
+def Equivalent (Q₁ : QuadraticForm R M₁) (Q₂ : QuadraticForm R M₂) :=
+  Nonempty (Q₁.Isometry Q₂)
+#align quadratic_form.equivalent QuadraticForm.Equivalent
+
+namespace Isometry
+
+variable {Q₁ : QuadraticForm R M₁} {Q₂ : QuadraticForm R M₂} {Q₃ : QuadraticForm R M₃}
+
+-- Porting note: was `Coe`
+instance : CoeOut (Q₁.Isometry Q₂) (M₁ ≃ₗ[R] M₂) :=
+  ⟨Isometry.toLinearEquiv⟩
+
+-- Porting note: syntaut
+#noalign quadratic_form.isometry.to_linear_equiv_eq_coe
+
+instance : CoeFun (Q₁.Isometry Q₂) fun _ => M₁ → M₂ :=
```

Comment:
We've been turning most of these into `EquivLike` or `FunLike` during the port.

Reply:
If you'd like to block this PR on that, can you explain what changes you need, exactly, or push them yourself?

Reply:
I replaced the `CoeFun` with an `EquivLike` and restored `coe_toLinearEquiv`. These things have caused issues in the past, and it was a bit of work to fix all the breakages in #3082 for example